arrow/compute/exprs/exec.go

```diff
@@ -613,7 +632,7 @@ func executeScalarBatch(ctx context.Context, input compute.ExecBatch, exp expr.E
 			result.Release()
```

Comment:
Do we need `result = nil` here? (Can we return `Release()`-ed `result`?)

Reply:
hmm, probably safer to make it `nil` you're right. just in case.